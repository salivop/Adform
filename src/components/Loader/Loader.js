import React from "react";
import CircularProgress from "@material-ui/core/CircularProgress";
import Typography from "@material-ui/core/Typography/Typography";

const Loader = () => {
  return (
    <React.Fragment>
      <div align="center">
        <CircularProgress />
      </div>
      <Typography align="center">Please wait...</Typography>
    </React.Fragment>
  );
};

export default Loader;
