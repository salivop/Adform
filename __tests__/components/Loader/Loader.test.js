import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import Loader from "src/components/Loader/Loader";

describe("Loader", () => {
  let wrapper;

  beforeEach(() => (wrapper = shallow(<Loader />)));

  it("should render correctly", () => {
    const component = wrapper;
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should have Fragment", () => {
    expect(wrapper.find("Fragment").length).toEqual(1);
  });

  it("should have Typography with styles", () => {
    expect(wrapper.find("WithStyles(ForwardRef(Typography))").length).toEqual(
      1
    );
    expect(
      wrapper.find("WithStyles(ForwardRef(Typography))").get(0).props
    ).toEqual(
      expect.objectContaining({
        align: "center"
      })
    );
  });

  it("should have div with styles", () => {
    expect(wrapper.find("div").length).toEqual(
      1
    );
    expect(
      wrapper.find("div").get(0).props
    ).toEqual(
      expect.objectContaining({
        align: "center"
      })
    );
  });

  it("should Typography have text", () => {
    expect(
      wrapper
        .find("WithStyles(ForwardRef(Typography))")
        .at(0)
        .props().children
    ).toEqual("Please wait...");
  });
});
