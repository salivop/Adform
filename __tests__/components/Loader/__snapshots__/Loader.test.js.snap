// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render correctly 1`] = `
<Fragment>
  <div
    align="center"
  >
    <WithStyles(ForwardRef(CircularProgress)) />
  </div>
  <WithStyles(ForwardRef(Typography))
    align="center"
  >
    Please wait...
  </WithStyles(ForwardRef(Typography))>
</Fragment>
`;
